File: backend/internal/statement/subject.go
package statement

import "app/internal/expression"

type Subject struct {
	expr expression.Expression
}

func NewSubject(expr expression.Expression) *Subject {
	return &Subject{
		expr: expr,
	}
}

func (subject *Subject) ToString() string {
	return subject.expr.PrettyString()
}

File: backend/internal/statements/subject.py
from backend.internal.expressions import Expression
from backend.internal.statements import Statement


class Subject(Statement):
    def __init__(self, expr: Expression) -> None:
        super().__init__()
        self._expr = expr

    def __eq__(self, other):
        return isinstance(other, Subject) and self._expr == other._expr

    def __repr__(self) -> str:
        return f"Subject({repr(self._expr)})"

    def __str__(self) -> str:
        return str(self._expr)

    @property
    def expr(self) -> Expression:
        return self._expr
